test/rules/invariant_booleans.dart

```diff
@@ -237,3 +237,13 @@ void bug372(bool foo) {
     // doSomethingElse();
   }
 }
+
+void bug337(int offset, int length) {
+  if (offset >= length) {
+    return;
+  }
+
+  offset++;
```

Comment:
You might want to introduce other variables so that you can check other operators.